base/Dockerfile

```diff
@@ -10,6 +10,7 @@ RUN groupadd -r socrata && useradd -r -g socrata socrata
 RUN groupadd -g 31415 metrics && usermod -a -G metrics socrata
 
 RUN DEBIAN_FRONTEND=noninteractive apt-get -y update && \
+  DEBIAN_FRONTEND=noninteractive apt-get -y dist-upgrade && \
```

Comment:
wont this update things like python, etc that we use to build our/run our products? if so, we may need to be more deliberate about what we update and when so as not to unexpectedly upgrade a service dependency.

Reply:
Yes and no.  On our `socrata/base` image, we pull in `ubuntu:trusty` which we've confirmed has updated system level packages.  The last time it was upgraded was on 1/23.  However, `socrata/runit` is a different story, it has not.

The thought is that we don't want to rely on remote parties to upgrade the images since it may not happen within the appropriate amount of time to remain compliant from a fedramp perspective.

I _think_ the only other option when considering doing periodic upgrades is the how?  Right now we are triggering a weekly rebuild of `socrata/base` and `socrata/runit` in Dockerhub, but this only builds based off the `Dockerfile`.  It also triggers a rebuild on new commits to the master branch for each `Dockerfile`.  If doing a weekly upgrade of the os files is an issue, we might be able to use Jenkins to pull the image down, run it, patch it with `dist-upgrade`, then push it up to Dockerhub.  This might run once a month?

Reply:
looking at where we refer to these base images it looks like (at least for ruby) we are ok.  With the introduction of the x.y.z versioned ruby images this last wee we explicitly install the ruby version we want so if we do that for all our other language types we are probably ok. We should confirm that explicit version setting is true for the other language types (elixir, java nodjs, python).

Reply:
I'd agree with you there.  I've looked at `socrata/python3`, `socrata/ruby`, and a couple of others.  It looks like we build from the base, which I expect.  We're building packages after the fact for specific ruby and python versions.  From the little projects I've handled, I've seen the python projects have the version number of libs in the `requirements.txt`.  Gem files seem to do the same.  Who can I reach out to for further confirmation?>